Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -297,8 +306,11 @@ protected override bool OnBackButtonPressed()
 
 		void InsertPageBefore(Page page, Page before)
 		{
+			if (page == null || before == null)
+				throw new ArgumentNullException($"Neither {nameof(page)} nor {nameof(before)} can be null.");
```

Comment:
This should be two separate checks, so the user knows _which_ of these two arguments was null.